crates/rspack_loader_swc/src/compiler.rs

```diff
@@ -346,16 +350,37 @@ impl SwcCompiler {
     }
   }
 
-  pub fn transform(&self, config: BuiltInput<impl Pass>) -> Result<Program, Error> {
+  pub fn transform(&self, config: BuiltInput<impl Pass>) -> Result<Program, miette::Report> {
     let program = config.program;
     let mut pass = config.pass;
 
     let program = self.run(|| {
       helpers::HELPERS.set(&self.helpers, || {
-        try_with_handler(self.cm.clone(), Default::default(), |handler| {
-          HANDLER.set(handler, || Ok(program.apply(&mut pass)))
-        })
-      })
+      let result = try_with_handler(self.cm.clone(), Default::default(), |_handler| {
+        let result = program.apply(&mut pass);
+        Ok(result)
+      });
+      match result {
+        Ok(v) => Ok(v),
+        Err(err) => {
+        let error_msg = match err.downcast_ref::<String>(){
+          Some(msg) => {
+            msg
+          },
+          None => "unknown error"
+        };
+        // FIXME: with_help has bugs, use with_help when diagnostic print is fixed
```

Comment:
ModuleBuildError will double print help msg if diagnostic contains help, which should be fixed